packages/docusaurus-theme-search-algolia/src/theme/SearchBar/index.js

```diff
@@ -5,135 +5,133 @@
+    importDocSearchModalIfNeeded().then(() => {
+      setIsOpen(true);
+    });
+  }, [importDocSearchModalIfNeeded, setIsOpen]);
 
-    loadAlgolia(needFocus);
-  });
+  const onClose = useCallback(() => {
+    setIsOpen(false);
+  }, [setIsOpen]);
 
-  const handleSearchInputPressEnter = useCallback((e) => {
-    if (!e.defaultPrevented && e.key === 'Enter') {
-      navigateToSearchPage(e.target.value);
-    }
-  });
+  useDocSearchKeyboardEvents({isOpen, onOpen, onClose});
 
   return (
-    <div className="navbar__search" key="search-box">
-      <div className={styles.searchWrapper}>
-        <span
-          aria-label="expand searchbar"
-          role="button"
-          className={clsx(styles.searchIconButton, {
-            [styles.searchIconButtonHidden]: isSearchBarExpanded,
-          })}
-          onClick={toggleSearchInput}
-          onKeyDown={toggleSearchInput}
-          tabIndex={0}
+    <>
+      <Head>
+        {/* This hints the browser that the website will load data from Algolia,
+        and allows it to preconnect to the DocSearch cluster. It makes the first
+        query faster, especially on mobile. */}
+        <link
+          rel="preconnect"
+          href={`https://${appId}-dsn.algolia.net`}
+          crossOrigin
         />
-
-        <input
-          id="search_input_react"
-          type="search"
-          placeholder="Search"
-          aria-label="Search"
-          className={clsx('navbar__search-input', styles.searchInput, {
-            [styles.searchInputExpanded]: isSearchBarExpanded,
-          })}
-          onMouseOver={handleSearchInput}
-          onFocus={handleSearchInput}
-          onBlur={handleSearchInputBlur}
-          onKeyDown={handleSearchInputPressEnter}
-          ref={searchBarRef}
-        />
-      </div>
-    </div>
+      </Head>
+
+      <DocSearchButton
+        onTouchStart={importDocSearchModalIfNeeded}
+        onFocus={importDocSearchModalIfNeeded}
+        onMouseOver={importDocSearchModalIfNeeded}
+        onClick={onOpen}
+      />
+
+      {isOpen &&
+        createPortal(
+          <DocSearchModal
+            appId={appId}
+            apiKey={apiKey}
+            indexName={indexName}
+            searchParameters={searchParameters}
+            onClose={onClose}
+            initialScrollY={window.scrollY}
+            navigator={{
+              navigate({suggestionUrl}) {
+                history.push(suggestionUrl);
+              },
+            }}
+            transformItems={(items) => {
+              return items.map((item) => {
+                // We transform the absolute URL into a relative URL.
+                // Alternatively, we can use `new URL(item.url)` but it's not
+                // supported in IE.
+                const a = document.createElement('a');
+                a.href = item.url;
+
+                return {
+                  ...item,
+                  url: `${a.pathname}${a.hash}`,
+                };
```

Comment:
This thing does not seem to work when we have a site baseUrl.

I think this is because we don't have a "baseUrl" header, and the url prefix is added manually to as a prefix to each link (not sure exactly why we do that, but it's probably hard to change without refactoring now)


```
config:
  baseUrl: '/build/', // because the production build is in website/built so it's convenient to test

yarn start

OR

yarn build:v2 && yarn serve website
open http://localhost:5000/build
```

As you can see, the site is served under /build, but the search links don't have the /build prefix

![image](https://user-images.githubusercontent.com/749374/86922765-64612a80-c12d-11ea-859c-8a131c8fdbc2.png)

You should be able to read the base url from the theme.

Note, there's already a `useBaseUrl()` in core to get some inspiration.

I think we'll need at some point a core hook like "const createAbsoluteUrl = useAbsoluteUrlCreator()" or something to be used in such cases, but we can do this later outside of this PR.

Reply:
Related PR:

https://github.com/facebook/docusaurus/pull/3033

API:

![image](https://user-images.githubusercontent.com/749374/86932228-5f09dd00-c139-11ea-961d-3e2e2803178e.png)

Going to merge it soon so that you can use it

Reply:
Thank you — done in 875c046deed29abcaf0177d296db67215b66b908.